### DISCUSSION TITLE
Python version

### DISCUSSION THREAD

**Discussion by rubenskuhl** [irrdnet/irrd#674] [Q&A] [answered]
Besides soon dropping support for Python 3.6, what Python versions are currently tested when building IRRD 4.x ?

**Comment by mxsasha** [comment]
Currently tested in CI is CPython 3.7 and 3.10, and PyPy 3.7 and 3.9. Probably will update it to CPython 3.11 once that is released.

For the future, I think we'll follow that versions are dropped in the first major (4.x+1) release [after that Python version is EOL](https://endoflife.date/python). Might deviate if really needed, but #604 showed that quickly becomes difficult.

**Comment by mxsasha** [comment, marked as answer]
Following up on this: 4.3 will support Python 3.7-3.11. As 3.7 is also close to EOL, and I maintain minor releases for a while, 4.4 will require 3.8 minimum.

The Python versions regularly tested are listed in (note there's CPython and PyPy):
https://github.com/irrdnet/irrd/blob/main/.circleci/config.yml#L17-L44

Before release, I run an extensive test with:
https://github.com/irrdnet/irrd/blob/main/.circleci/config.yml#L17-L50
